MAPL standalone build appears to be broken

I checked out develop and tried to build MAPL on Discover.  So checkout the MAPL repo, go there checkout develop, then mepo clone, the source g5_modules, then build.
Note I've also tried this with main and v2.7.0
I'm getting a slew of errors when running the cmake command:
```
-- Detecting C compile features
-- Detecting C compile features - done
Traceback (most recent call last):
  File "/discover/swdev/gmao_SIteam/mepo/mepo", line 17, in <module>
    main()
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/main.py", line 6, in main
    command.run(args)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/command/command.py", line 11, in run
    cmd_module.run(args)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/command/init/init.py", line 17, in run
    allcomps = MepoState.initialize(args.config,style)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/state/state.py", line 58, in initialize
    raise StateAlreadyInitializedError('mepo state already exists')
state.exceptions.StateAlreadyInitializedError: mepo state already exists
fatal: destination path './@ESMA_env' already exists and is not an empty directory.

Traceback (most recent call last):
  File "/discover/swdev/gmao_SIteam/mepo/mepo", line 17, in <module>
    main()
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/main.py", line 6, in main
    command.run(args)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/command/command.py", line 11, in run
    cmd_module.run(args)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/command/clone/clone.py", line 68, in run
    git.clone(version,recurse)
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/repository/git.py", line 45, in clone
    shellcmd.run(cmd.split())
  File "/gpfsm/dswdev/gmao_SIteam/mepo/mepo.d/utilities/shellcmd.py", line 15, in run
    result.check_returncode()
  File "/usr/local/other/python/GEOSpyD/4.9.2_py3.9/2021-05-05/lib/python3.9/subprocess.py", line 460, in check_returncode
    raise CalledProcessError(self.returncode, self.args, self.stdout,
subprocess.CalledProcessError: Command '['git', 'clone', '--branch', 'v3.2.1', '--quiet', 'git@github.com:GEOS-ESM/ESMA_env.git', './@ESMA_env']' returned non-zero exit status 128.
CMake Error at CMakeLists.txt:37 (include):
  include could not find load file:

    esma


CMake Error at CMakeLists.txt:45 (find_package):
  By not providing "FindFLAP.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "FLAP", but
  CMake did not find one.

  Could not find a package configuration file provided by "FLAP" with any of
  the following names:

    FLAPConfig.cmake
    flap-config.cmake

  Add the installation prefix of "FLAP" to CMAKE_PREFIX_PATH or set
  "FLAP_DIR" to a directory containing one of the above files.  If "FLAP"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/bmauer/swdev/models/mapl_fix_regrid/mapl_bare/MAPL/build/CMakeFiles/CMakeOutput.log".
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Set required CMake Version to 3.17
+- Update `components.yaml`
+  - ESMA_cmake v3.4.2 (to match GEOSgcm)
 
 ### Fixed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Removed
 ### Added
 ### Changed
+
+- Set required CMake Version to 3.17
+
 ### Fixed
+
+- Allow MAPL to build if subrepos are cloned with any mepo style
+  (prefix, postfix, naked)
 
 ## [2.7.0] - 2021-05-25
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.13)
+cmake_minimum_required (VERSION 3.17)
 cmake_policy (SET CMP0053 NEW)
 cmake_policy (SET CMP0054 NEW)
 
@@ -7,10 +7,19 @@ project (
   VERSION 2.7.0
   LANGUAGES Fortran CXX C)  # Note - CXX is required for ESMF
 
-if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/ESMA_cmake)
-  list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/ESMA_cmake")
-  include (esma)
-endif ()
+# mepo can now clone subrepos in three styles
+set (ESMA_CMAKE_DIRS
+  ESMA_cmake
+  @ESMA_cmake
+  ESMA_cmake@
+  )
+
+foreach (dir IN LISTS ESMA_CMAKE_DIRS)
+  if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/${dir})
+    list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/${dir}")
+    include (esma)
+  endif ()
+endforeach ()
 
 # build as standalone project
 if (NOT COMMAND esma)
@@ -22,19 +31,17 @@ if (NOT COMMAND esma)
   if (SKIP_MEPO)
     message (FATAL_ERROR "ESMA not found")
   else ()
-     set (MEPO_INIT_COMMAND mepo init)
-     execute_process (
-        COMMAND ${MEPO_INIT_COMMAND}
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-        )
-
-     set (MEPO_CLONE_COMMAND mepo clone)
-     execute_process (
-        COMMAND ${MEPO_CLONE_COMMAND}
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-        )
-    list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/ESMA_cmake")
-    include (esma)
+    set (MEPO_CLONE_COMMAND mepo clone)
+    execute_process (
+      COMMAND ${MEPO_CLONE_COMMAND}
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      )
+    foreach (dir IN LISTS ESMA_CMAKE_DIRS)
+      if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/${dir})
+        list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/${dir}")
+        include (esma)
+      endif ()
+    endforeach ()
     option (SKIP_MEPO "Set to skip mepo steps" ON)
   endif ()
 

--- a/components.yaml
+++ b/components.yaml
@@ -11,7 +11,7 @@ ESMA_env:
 ESMA_cmake:
   local: ./ESMA_cmake
   remote: ../ESMA_cmake.git
-  tag: v3.4.0
+  tag: v3.4.2
   develop: develop
 
 ecbuild:

